### PR TITLE
Allow float gauges

### PR DESCRIPTION
**Pull request by messiaen**
@vthiery Currently only int values can be given for gauges due to the functions definitions.  I'd like to be able to send floats.

### CODE REVIEW

File: include/cpp-statsd-client/StatsdClient.hpp
Comment (kevinkreiser) [inline include/cpp-statsd-client/StatsdClient.hpp:304]: @messiaen rather than add `to_string` everywhere could we just `template` the send function so it could take either float or int?

File: include/cpp-statsd-client/StatsdClient.hpp
Comment (kevinkreiser) [inline include/cpp-statsd-client/StatsdClient.hpp:170]: same point as below, why not use templates?

File: tests/testStatsdClient.cpp
Comment (kevinkreiser) [inline tests/testStatsdClient.cpp:118]: my biggest question about this PR is about precision. single precision floats are notorious for storing essentially garbage noise after the 6th (or so) digit. converting and sending these bytes seems wasteful. maybe we should be using a fixed precision for floating point numbers? or maybe the extra noise doesnt bother anyone else?

Comment (messiaen) [reply tests/testStatsdClient.cpp:118]: Added a precision config value in StatusdClient.

File: include/cpp-statsd-client/StatsdClient.hpp
Comment (messiaen) [reply include/cpp-statsd-client/StatsdClient.hpp]: Using template now.

File: include/cpp-statsd-client/StatsdClient.hpp
Comment (kevinkreiser) [inline include/cpp-statsd-client/StatsdClient.hpp:191]: a couple notes on style:

it seems that your formatter is doing some strange things with the white space which is makign the diff hard to read. we are using clangformat here. also notice that we use `hungarian` notation, so that variable should be `m_guagePrecision` since its a member variable. also, i would suggest not using `this->` its superfluous (and looks like java :trollface: )

File: include/cpp-statsd-client/StatsdClient.hpp
Comment (kevinkreiser) [inline include/cpp-statsd-client/StatsdClient.hpp:225]: ha good catch!

File: include/cpp-statsd-client/StatsdClient.hpp
Comment (kevinkreiser) [inline include/cpp-statsd-client/StatsdClient.hpp:143]: precision is a member variable so you dont need to change this method signature, the method has already access to the member variable

File: include/cpp-statsd-client/StatsdClient.hpp
Comment (messiaen) [reply include/cpp-statsd-client/StatsdClient.hpp]: - Ran clang-format again.
- Changed instance variable name.

**Review by kevinkreiser — APPROVED** [review]
the code looks good to me but i defer to @vthiery. also we need to allow CI to run too just to make sure

File: include/cpp-statsd-client/StatsdClient.hpp
Comment (kevinkreiser) [inline include/cpp-statsd-client/StatsdClient.hpp:92]: nothing changed here but there is still a diff, i guess you must have a different clangformat version or a different default line width? any chance you can clean up these whitespace changes so that the diff is just focused on actual changes?

File: include/cpp-statsd-client/StatsdClient.hpp
Comment (vthiery) [reply include/cpp-statsd-client/StatsdClient.hpp:221]: Indeed :)
The same can be done to `decrement` and `increment`.

File: include/cpp-statsd-client/StatsdClient.hpp
Comment (vthiery) [inline include/cpp-statsd-client/StatsdClient.hpp:189]: You can do like `m_prefix` and `m_sender` line 184:
```cpp
: m_prefix(detail::sanitizePrefix(prefix)), m_sender(new UDPSender{host, port, batchsize, sendInterval}), m_gaugePrecision(gaugePrecision)
```

File: tests/testStatsdClient.cpp
Comment (vthiery) [inline tests/testStatsdClient.cpp:120]: We should check that the precision is satisfied:
```suggestion
        // Record a gauge "titifloat" to 123.456789 with precision 3
        client.gauge("titifloat", 123.456789);
        throwOnError(client);
        expected.emplace_back("sendRecv.titifloat:123.457|g");
```

**Review by vthiery — CHANGES_REQUESTED** [review]
Thanks for the PR! Let's just tweak the test a bit (or add one, as you wish), fix the style and then it should be ok :+1:

**Comment by vthiery** [comment]
> the code looks good to me but i defer to @vthiery. also we need to allow CI to run too just to make sure

Thanks for the review @kevinkreiser ! I allowed the CI to run :+1:

File: include/cpp-statsd-client/StatsdClient.hpp
Comment (vthiery) [inline include/cpp-statsd-client/StatsdClient.hpp:69]: We could also add a quick note about the setting float precision (a line or two is enough) in the README, like we did for the other parameters.

**Comment by messiaen** [comment]
Thanks @vthiery and @kevinkreiser! 
- Fixed formatting
- Added note about gaugePrecision in code examples in README
- tweaked test.

File: tests/testStatsdClient.cpp
Comment (messiaen) [reply tests/testStatsdClient.cpp]: Made this change manually.

**Review by vthiery — APPROVED** [review]
LGTM
@kevinkreiser - want to have another look before I merge it?

**Comment by dmitrifried** [comment]
Hi @vthiery, are you planning on making a new release version with this commit? Right now our cmake dependency will have to point to the commit hash to get the float guages.

**Comment by vthiery** [comment]
> Hi @vthiery, are you planning on making a new release version with this commit? Right now our cmake dependency will have to point to the commit hash to get the float guages.

Hi @dmitrifried 
I'll release a v2 today or tomorrow. I just need to prepare a quick migration guide since we've been making quite a lot of changes lately.

**Comment by vthiery** [comment]
> > Hi @vthiery, are you planning on making a new release version with this commit? Right now our cmake dependency will have to point to the commit hash to get the float guages.
> 
> Hi @dmitrifried
> I'll release a v2 today or tomorrow. I just need to prepare a quick migration guide since we've been making quite a lot of changes lately.

@dmitrifried I just release [v2.0.0](https://github.com/vthiery/cpp-statsd-client/releases/tag/v2.0.0)

**Comment by dmitrifried** [comment]
@vtheiry Thanks for the quick work!

**Pull request merged**
